In [8]:
install.packages("tidyverse")
library(tidyverse)
set.seed(42)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [9]:
# Generating Dataset With 4 Treatment Groups
anova_data <- tibble(
  patient_ID = 1:120,
  drug_group = factor(rep(c("Drug_A" , "Drug_B" , "Drug_C" , "Placebo") , each = 30)),
  bp_drop = c(rnorm(30, mean = 18, sd = 4) , # Drug A Works Great
  rnorm(30, mean = 12, sd = 4), # Drug B Works Okay
  rnorm(30, mean = 5, sd = 4), # Drug C Is Weak
  rnorm(30, mean = 2, sd = 4) # Placebo)
  )
)

In [10]:
# Performing NOVA (Analysis Of Variance)
anova_result <- aov(bp_drop ~ drug_group, data = anova_data)
summary(anova_result)

             Df Sum Sq Mean Sq F value Pr(>F)    
drug_group    3   4570  1523.2   87.45 <2e-16 ***
Residuals   116   2020    17.4                   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

In [11]:
TukeyHSD(anova_result)

  Tukey multiple comparisons of means
    95% family-wise confidence level

Fit: aov(formula = bp_drop ~ drug_group, data = anova_data)

$drug_group
                     diff        lwr        upr     p adj
Drug_B-Drug_A   -6.761981  -9.570835  -3.953128 0.0000000
Drug_C-Drug_A  -12.511270 -15.320124  -9.702417 0.0000000
Placebo-Drug_A -16.349706 -19.158560 -13.540853 0.0000000
Drug_C-Drug_B   -5.749289  -8.558143  -2.940435 0.0000028
Placebo-Drug_B  -9.587725 -12.396579  -6.778871 0.0000000
Placebo-Drug_C  -3.838436  -6.647290  -1.029582 0.0029677


### Project Summary:

1. Purpose:
The Objective Of This Analysis Was To Evaluate The Effectiveness Of Three Different Drug Groups (Drug_A, Drug_B, Drug_C) Compared To A Control Group (Placebo)on Lowering Blood Pressure (bp_drop).

2. ANOVA Results:
The One-Way ANOVA Test Revealed A Highly Significant Difference Between The Treatment Groups (F(3, 116) = 87.45, p < 2e-16). This Indicates That The Choice Of Drug Significantly Impacts Blood Pressure Reduction.

3. Post-Hoc Tukey HSD Results:
The Pairwise Comparisons Showed That All groups Are Statistically Significantly Different From One Another (All Adjusted p-values < 0.05).
- Drug_A Showed The Highest Efficiency In Reducing Blood Pressure.
- Drug_B Performed Moderately Well, Significantly Better That Drug_C And Placebo, But Less Than Drug_A.
- Drug_C Had A Weak Effect, But It Was Still Statistically Better That The Placebo Group.

Conclusion:
The Statistical Hierarchy Of Effectiveness Is: Drug_A > Drug_B > Drug_C > Placebo.